helix-term/src/commands/typed.rs

```diff
@@ -1606,6 +1606,100 @@ fn run_shell_command(
     Ok(())
 }
 
+fn read_file_info_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    let (view, doc) = current!(cx.editor);
+
+    ensure!(!args.is_empty(), "file name is expected");
+
+    let filename = args.get(0).unwrap();
+    let path = PathBuf::from(filename.to_string());
+    if path.exists() {
+        match std::fs::File::open(path) {
+            Ok(file) => {
+                let mut reader = BufReader::new(file);
+
+                match from_reader(&mut reader, doc.encoding()) {
+                    Ok(contents) => {
+                        let contents = Tendril::from(contents);
+                        let selection = doc.selection(view.id);
+                        let transaction = Transaction::insert(doc.text(), selection, contents);
+                        doc.apply(&transaction, view.id);
+                    }
+                    Err(error) => cx
+                        .editor
+                        .set_error(format!("error reading file: {}", error)),
+                }
+            }
+            Err(error) => cx
+                .editor
+                .set_error(format!("error reading file: {}", error)),
+        }
+    } else {
+        cx.editor
+            .set_error(format!("file doesn't exist: {}", filename));
+    }
+
+    Ok(())
+}
+
+/// Stripped down version of [`helix_view::document::from_reader`] which is adapted to use encoding_rs::Decoder::read_to_string
+fn from_reader<R: std::io::Read + ?Sized>(
```

Comment:
This duplicates a lot of code between this function and `document::from_reader`. Can you factor out the shared parts and share the code?

Reply:
I agree. I was following @kirawi's advice on this: https://github.com/helix-editor/helix/pull/3966#discussion_r979462663

I can try working something out at the expense of dragging this PR further, or that work can be the subject of another PR. I am easy either way. It's all up to the maintainers at this point.

Reply:
This can use `read_to_string` from document now: https://github.com/helix-editor/helix/blob/636c91c76b2855a4ac58b3a030a9e45f88eb7502/helix-view/src/document.rs#L474-L477